### ISSUE TITLE
Buggy control flow in calc_HEX_heating (substation model)

### ISSUE THREAD

**Issue by daren-thomas** [architecture-building-systems/CityEnergyAnalyst#1668]
The variable `Flag` (FIXME: rename this to PEP 8 and convey variable meaning in name) is set to `False` after the first iteration of the `while` loop. This begs the question, is this supposed to be a simple `if`? Or is the control flow broken here?

```
def calc_HEX_heating(Q, UA, thi, tco, tci, cc):
   # [...]

    if Q > 0.0:
        dT_primary = tco - tci if tco != tci else 0.0001  # to avoid errors with temperature changes < 0.001
        eff = [0.1, 0]
        Flag = False
        tol = 0.00000001
        while abs((eff[0] - eff[1]) / eff[0]) > tol:
            if Flag == True:
                eff[0] = eff[1]
            else:
                cmin = cc * (dT_primary) / ((thi - tci) * eff[0])
            if cmin < cc:
                ch = cmin
                cmax = cc
            else:
                ch = cmin
                cmax = cmin
                cmin = cc
            cr = cmin / cmax
            NTU = UA / cmin
            eff[1] = calc_shell_HEX(NTU, cr)
            cmin = cc * (dT_primary) / ((thi - tci) * eff[1])
            tho = thi - eff[1] * cmin * (thi - tci) / ch
            Flag = True

        tho = tho - 273
    else:
        tho = 0
        ch = 0
    return np.float(tho), np.float(ch)
```

**Comment by daren-thomas** [comment]
actually, it's not that tough. @BhargavaKrishnaSreepathi were able to work it out in a pair-programming session. basically, the `Flag` variable here has the meaning "first run through the loop", which can totally be replaced by moving that to outside of the loop - except, it doesn't even make sense there, since `eff` is right before the loop and has a known state. The PR #1622 contains the refactored version.

**Comment by BhargavaKrishnaSreepathi** [comment]
fixed in PR #1678

**Comment by daren-thomas** [comment]
PR #1678 has not been accepted yet.
